### ISSUE TITLE
[Feature Request] Request for weighted sampling in tf.data.Dataset 

### ISSUE THREAD

**Issue by niffler92** [tensorflow/tensorflow#15999]
Are there any methods that we can sample with weights without pre-weighting the dataset as the input in Dataset API?

**Comment by mrry** [comment]
The only sampling-specific method in the API is [`tf.contrib.data.rejection_resample()`](https://www.tensorflow.org/api_docs/python/tf/contrib/data/rejection_resample). If that doesn't work for your application, can you specify what sampling operation you'd like? Like `rejection_resample()` it may be possible to build it out of existing parts, like `Dataset.filter()` and `tf.contrib.data.scan()`.

**Comment by niffler92** [comment]
I think explicit methods for giving weights corresponding to the label would be highly useful. e.g. Dataset.resample(label, weight)

**Comment by tensorflowbutler** [comment]
The original poster has replied to this issue after the stat:awaiting response label was applied.

**Comment by mrry** [comment]
The `tf.contrib.data.rejection_resample()` transformation allows you to specify a mapping from `Dataset` elements to an integer class, and a vector of probabilities representing the target distribution (indexed by class) for the resampling. Can you be more specific about what a `Dataset.resample()` method would do in addition to this?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jperl** [comment]
Is it possible to sample rows from our dataset using a custom distribution like `[tf.distributions.Exponential](https://www.tensorflow.org/api_docs/python/tf/distributions/Exponential)`?

**Comment by lytkarinskiy** [comment]
Hello all,
I'm also interested if it's possible to batch elements from Dataset with distribution, so to pick last elements more often than elements in the begin of dataset.

I'm talking about numeric dataset made from tensor_slices with initial shape 1000*10*3

In this case depending on distribution probability we'll take elements with index [999,:,:] more often than elements with index [1,:,:].

**Comment by hyeokhyen** [comment]
Why is this closed? Weighted sampling of mini batch is one of very useful feature in pytorch, especially when suffering under have label imbalance.

**Comment by shreyaskamathkm** [comment]
Is there a workaround to create a weighted sampler?
